sagemaker-clarify/fairness_and_explainability/fairness_and_explainability_boto3.ipynb

```diff
@@ -0,0 +1,1541 @@
+       "      <td>7</td>\n",
+       "      <td>2</td>\n",
+       "      <td>5</td>\n",
+       "      <td>0</td>\n",
+       "      <td>2</td>\n",
+       "      <td>1</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>40</td>\n",
+       "      <td>38</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>0</td>\n",
+       "      <td>28</td>\n",
+       "      <td>2</td>\n",
+       "      <td>338409</td>\n",
+       "      <td>9</td>\n",
+       "      <td>13</td>\n",
+       "      <td>2</td>\n",
+       "      <td>9</td>\n",
+       "      <td>5</td>\n",
+       "      <td>2</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>0</td>\n",
+       "      <td>40</td>\n",
+       "      <td>4</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   Target  Age  Workclass  fnlwgt  Education  Education-Num  Marital Status  \\\n",
+       "0       0   39          5   77516          9             13               4   \n",
+       "1       0   50          4   83311          9             13               2   \n",
+       "2       0   38          2  215646         11              9               0   \n",
+       "3       0   53          2  234721          1              7               2   \n",
+       "4       0   28          2  338409          9             13               2   \n",
+       "\n",
+       "   Occupation  Relationship  Ethnic group  Sex  Capital Gain  Capital Loss  \\\n",
+       "0           0             1             4    1          2174             0   \n",
+       "1           3             0             4    1             0             0   \n",
+       "2           5             1             4    1             0             0   \n",
+       "3           5             0             2    1             0             0   \n",
+       "4           9             5             2    0             0             0   \n",
+       "\n",
+       "   Hours per week  Country  \n",
+       "0              40       38  \n",
+       "1              13       38  \n",
+       "2              40       38  \n",
+       "3              40       38  \n",
+       "4              40        4  "
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "training_data.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lastly, let's upload the data to S3."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker.s3 import S3Uploader\n",
+    "from sagemaker.inputs import TrainingInput\n",
+    "\n",
+    "train_uri = S3Uploader.upload(\"train_data.csv\", \"s3://{}/{}\".format(bucket, prefix))\n",
```

Comment:
General: Pass the sagemaker_session object to the `sagemaker_session` parameter of the upload method, e.g.,

```
train_uri = S3Uploader.upload(
    local_path="train_data.csv",
    desired_s3_uri="s3://{}/{}".format(bucket, prefix),
    sagemaker_session=sagemaker_session,
)
```

Otherwise the upload method will have to [create a new session](https://github.com/aws/sagemaker-python-sdk/blob/7faf9ce252a7694cac63be3a8ec8986776276708/src/sagemaker/s3.py#L82), which is a waste.